components/proxy/src/test/java/com/hotels/styx/infrastructure/FileBackedRegistryTest.java

```diff
@@ -41,14 +75,54 @@ public void calculatesTheDifferenceBetweenCurrentAndNewResources() {
         assertThat(changes.toString(), is(expected.toString()));
     }
 
-    private BackendService backendService(String id, int port) {
+    @Test
+    public void logsFileChangeWhenChangeListenerThrowsAnExceptionDuringReload() throws IOException, TimeoutException {
+        log = new LoggingTestSupport(FileBackedRegistry.class);
+        tempFile = createTempFile("origins-", ".txt");
+
+        write(tempFile, "original_test_content".getBytes());
```

Comment:
Here you have to construct temporary files for testing purposes because code to read origins file (*readFile()*, *readFileAsString()*) is embedded inside *FileBackedRegistry*.

Please extract file reading out into its own class. Then use a mock in its place to get rid of temporary files.

I know this PR is just for adding instrumentation, but lets use the opportunity to add a small improvement. There is a chance it will stick around for awhile.